content/purgatorio/launch-and-mantain-the-system/monitoring-usage-and-behavior.md

```diff
@@ -1,5 +1,142 @@
+
+    - As a corollary, looking at overall change in the confidence of outputs. E.g., a drop in confidence values is possibly a concerning sign. (Of course, vanilla NNs are also prone to be highly overconfident on new/OOD data, so you need to be cautious; if this is really critical, could look at different NN variants that generate better confidence assessments.)
+
+    - Consider whether training a model w/ out-of-domain detector (OOD) makes sense or not.
+
+    See: 
+    - [How to determine the confidence of a neural network prediction?](https://stats.stackexchange.com/questions/247551/how-to-determine-the-confidence-of-a-neural-network-prediction)
+
+- **Human evaluation:**
+
+    - Not really going to be avoidable, but you can use a lot of the above to narrow down what people look at. Have people look at the examples which are most alarming, based on the checks you have in place. Over time, we can better calibrate what alerts are of concern.
+
+    - A related option is to use a calibrated model of confidence/estimated error rate to inform what people look at. If you pick up the lowest 1% confidence and the internal estimated error rate is 5%, then if you audit that 1% and see > 5% error rate, that could be a concern. 
+
+    See: 
+    - [Comparing Automatic and Human Evaluation of Local Explanations for Text Classification](https://www.aclweb.org/anthology/N18-1097.pdf)
+
+
+## Monitoring the Usage
+
+If monitoring the behavior of the model can be technically hard, you should also be sure that your users are leveraging the model in the correct way. 
+
+With "users" we can refer to _everything that consumes the output of the ML model_, it can be a system, a human, or an ensemble of systems and humans.
+
+::: tip 
+If you're serving your model through an API (recommended way), you can refer to the API monitoring best practices in general (not specific for Machine Learning).
+:::
+
+See: 
+
+- [What's the best way to monitor your REST API?](https://stackoverflow.com/questions/28760946/whats-the-best-way-to-monitor-your-rest-api)
+- [Measure Anything, Measure Everything](https://codeascraft.com/2011/02/15/measure-anything-measure-everything/)
+
+The major issue you can encounter when dealing with people is that they choose to not use the ML model. 
+
+This can happen for a variety of reasons, maybe they don't have _confidence enough_ in the system, or they don't understand _how to_ use it.
+
+Take a look at this awesome guide from Google's engineers:
+
+> The People + AI Guidebook was written to help user experience (UX) professionals and product managers follow a human-centered approach to AI.
+
+- [People + AI Guidebook](https://pair.withgoogle.com/guidebook)
+
+This detailed resources can get you started about the following topics: 
+
+- User Needs + Defining Success
+- Data Collection + Evaluation
+- Mental Models
+- Explainability + Trust
+- Feedback + Control
+- Errors + Graceful Failure
+
```

Comment:
Something to add to the monitor the usage, from the APIs it can become easier to filter down the lower or inner aspects of the models' functionality/features and these should also be tracked, so then you know if you make your next model, what to give importance to. This is crucial and useful information for the next set of features to engineer for the next iteration of the model.

_Production data -> (amended features) -> (amended algorithm + hyperparams) -> new more efficient model_

Reply:
I find this is too specific, but very good point, maybe we can insert this in the feature engineering part

Reply:
Tracking feature importances generated by a model (many algorithms generate such a list via method call i.e.`feature_importance()` or `coef`). If such functionality is absent then a custom methods need to be developed.

These kinds of tips can be placed into the best practices guide.

Reply:
Tracking confusion matrix and metrics like these are super important.

Reply:
TODO: Cross-references between this guide and best practices guide and other guides is super important.